Mage.Sets/src/mage/cards/o/OhabiCaleria.java

```diff
@@ -0,0 +1,102 @@
+package mage.cards.o;
+
+import mage.MageInt;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.costs.mana.GenericManaCost;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.common.DoIfCostPaid;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.abilities.effects.common.continuous.UntapAllDuringEachOtherPlayersUntapStepEffect;
+import mage.abilities.keyword.ReachAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.constants.Zone;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+
+import java.util.UUID;
+
+/**
+ * @author PurpleCrowbar
+ */
+public final class OhabiCaleria extends CardImpl {
+
+    private static final FilterCreaturePermanent filter = new FilterCreaturePermanent("Archers you control");
```

Comment:
This needs to be `FilterControlledCreaturePermanent` if you're using the common ability class. And you can put the Subtype right in the constructor for both `FilterCreaturePermanent` and `FilterControlledCreaturePermanent`.

Reply:
Done. However, I needed an additional filter for the correct message to display in the context of the third ability

Reply:
That's fine.